app/services/upstream_profile_remover.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+# Service for removing upstream profiles
+class UpstreamProfileRemover
+  class << self
+    def run!
+      Profile.canonical(false)
+             .joins(:parent_profile)
+             .joins(:test_results)
```

Comment:
What do we need test results for?

Reply:
That was the requirement from PM, check the jira ticket